### PR TITLE
PCUI Examples

### PR DESCRIPTION
**Pull request by willeastcott**
Added a set of simple examples for PCUI. Examples are:

* **Minimal.** Examples average about 50 lines each.
* **Single page.** No external dependencies beyond loading PCUI from [unpkg.com](unpkg.com).
* **ES Module-based.** Modern JS with ES Modules and Import Maps. No need for a build step using Rollup/Webpack/etc.

I've also added a basic `index.html` which uses PCUI to create a simple examples browser type application:

pcui-examples

For PCUI developers, the examples are good for:

* Debugging/developing individual elements/modules. By default, examples load the last NPM release of PCUI (via unpkg.com). To run against a local PCUI build, simply edit the import maps to relative paths.
* Sanity checking PCUI before generating a new release.

### CODE REVIEW

File: examples/colorpicker.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    <script async src="https://unpkg.com/es-module-shims/dist/es-module-shims.js"></script>
+    <script type="importmap">
+        {
+            "imports": {
+                "@playcanvas/observer": "https://unpkg.com/@playcanvas/observer/dist/index.mjs",
+                "@playcanvas/pcui": "https://unpkg.com/@playcanvas/pcui/dist/module/src/index.mjs",
+                "@playcanvas/pcui/styles": "https://unpkg.com/@playcanvas/pcui/styles/dist/index.mjs"
+            }
+        }
+    </script>
```
Comment (slimbuck) [inline examples/colorpicker.html:40]: Don't think this is meant to be here.

File: examples/index.html [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    <script async src="https://unpkg.com/es-module-shims/dist/es-module-shims.js"></script>
+    <script type="importmap">
+        {
+            "imports": {
+                "@playcanvas/observer": "https://unpkg.com/@playcanvas/observer/dist/index.mjs",
+                "@playcanvas/pcui": "https://unpkg.com/@playcanvas/pcui/dist/module/src/index.mjs",
+                "@playcanvas/pcui/styles": "https://unpkg.com/@playcanvas/pcui/styles/dist/index.mjs"
+            }
+        }
+    </script>
```
Comment (slimbuck) [inline examples/index.html:53]: Remove?

**Review by slimbuck — COMMENTED** [review]
Since all the examples appear to have the same HTML head, I wonder whether it wouldn't be better for each example to be just the script in each .html->body->script and do away with the duplication?

**Comment by willeastcott** [comment]
@slimbuck I know the duplication isn't ideal, but a developer can open any example and see how everything works straight away. No jumping between css/js/html files. At first, I did have the css in a common.css file but I moved it back into the html files. If you contrast this with the PlayCanvas Engine Examples Browser, there it's not obvious how to set up a project from scratch. Here, nothing is hidden and I kind of really like that. But totally open to debate on this as there's definitely no solution that is perfect. 😄

File: examples/colorpicker.html (hunk 1 above)
Comment (willeastcott) [reply examples/colorpicker.html:40]: Whoops. Fixed! Thanks.

File: examples/index.html (hunk 2 above)
Comment (willeastcott) [reply examples/index.html]: Fixed.

**Review by slimbuck — APPROVED** [review]

**Pull request merged**